src/main/kotlin/uk/gov/gdx/datashare/services/PrisonerLookupService.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.gdx.datashare.services
+
+import com.amazonaws.xray.spring.aop.XRayEnabled
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.gdx.datashare.enums.Sex
+import uk.gov.gdx.datashare.models.PrisonerDetails
+
+@Service
+@XRayEnabled
+class PrisonerLookupService(
+  private val prisonerApiService: PrisonerApiService,
+  private val objectMapper: ObjectMapper,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getEnrichedPayload(
+    prisonerNumber: String,
+    enrichmentFields: List<String>,
+  ): PrisonerDetails? {
+    val allEnrichedData = prisonerApiService.findPrisonerById(prisonerNumber)
+      ?.let {
+        PrisonerDetails(
+          prisonerNumber = prisonerNumber,
```

Comment:
I assume the prisonerNumber is guaranteed never to change and "redirect"?

Reply:
I assume so

Reply:
correct